packages/customize-widgets/src/components/sidebar-block-editor/sidebar-adapter.js

```diff
@@ -82,30 +90,47 @@ export default class SidebarAdapter {
 		};
 	}
 
-	trigger( event ) {
+	getWidgets() {
+		return this.history[ this.historyIndex ];
+	}
+
+	_emit( ...args ) {
 		for ( const callback of this.subscribers ) {
-			callback( event );
+			callback( ...args );
 		}
 	}
 
-	_handleSettingChange( newWidgetIds, oldWidgetIds ) {
-		const addedWidgetIds = difference( newWidgetIds, oldWidgetIds );
-		const removedWidgetIds = difference( oldWidgetIds, newWidgetIds );
+	_getWidgetIds() {
```

Comment:
It's slightly confusing that `getWidgets()` and `_getWidgetIds()` are named similarly but the former returns "adapted" data and the returns data from the underlying setting. I'd consider not having `_getWidgetIds()` and instead using `this.setting.get()` everywhere. It's more verbose but, in my opinion, less ambiguous.

Reply:
I tried that but I thought `this.setting.get()` isn't very clear what it would return.